File: Serenity.Script.UI/FilterPanel/FilteringTypes/IFiltering.cs
using jQueryApi;
using Serenity.Data;
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    public interface IFiltering
    {
        PropertyItem Field { get; set; }
        jQueryObject Container { get; set; }
        FilterOperator Operator { get; set; }
        void CreateEditor();
        CriteriaWithText GetCriteria();
        List<FilterOperator> GetOperators();
        void LoadState(object state);
        object SaveState();
    }

    [Imported, Serializable]
    public class CriteriaWithText
    {
        public BaseCriteria Criteria { get; set; }
        public string DisplayText { get; set; }
    }
}

File: Serenity.Web/Scripts/serenity/CoreLib/UI/Filtering/IFiltering.ts
declare namespace Serenity {
    interface IFiltering {
        createEditor(): void;
        getCriteria(displayText: any): any[];
        getOperators(): FilterOperator[];
        loadState(state: any): void;
        saveState(): any;
        get_field(): PropertyItem;
        set_field(value: PropertyItem): void;
        get_container(): JQuery;
        set_container(value: JQuery): void;
        get_operator(): FilterOperator;
        set_operator(value: FilterOperator): void;
    }
}